lib/pbench/test/unit/server/test_requests.py

```diff
@@ -1,14 +1,22 @@
 import os
+import socket
 
 import pytest
-import socket
-from werkzeug.utils import secure_filename
+import responses
+
 from pathlib import Path
+from werkzeug.utils import secure_filename
+
+
+@pytest.fixture
+def mocked_responses():
+    with responses.RequestsMock() as rsps:
+        yield rsps
```

Comment:
Hmm; I'm using the `requests_mock` fixture from `requests` for the same purpose, and I built up a `query_helper` fixture around it to help with setting up the mock, making the call, and checking results so I could easily pop off good/bad tests.

https://github.com/distributed-system-analysis/pbench/pull/1914/commits/14650fcac67f0225dfe4215306df6aef4fbde0ad#diff-94940849d430f436eeb18310e1a882f5c318d16637dc5097057d9eca1d2172ad (although it's gotten a bit thicker in the more recent commits).

I could probably just swap in your fixture, but I have no idea what the tradeoffs really are.

I had considered adding mock tests to the passthrough APIs, but decided to put that off, luckily, since you're doing it.

Reply:
From what I understand, `responses` handles all that work, but I might be missing some subtlety of that work.